contrib/cryptomb/private_key_providers/source/cryptomb_private_key_provider.cc

```diff
@@ -681,11 +681,10 @@ void CryptoMbPrivateKeyMethodProvider::unregisterPrivateKeyMethod(SSL* ssl) {
 CryptoMbPrivateKeyMethodProvider::CryptoMbPrivateKeyMethodProvider(
     const envoy::extensions::private_key_providers::cryptomb::v3alpha::
         CryptoMbPrivateKeyMethodConfig& conf,
-    Server::Configuration::TransportSocketFactoryContext& factory_context, IppCryptoSharedPtr ipp)
-    : api_(factory_context.serverFactoryContext().api()),
-      tls_(ThreadLocal::TypedSlot<ThreadLocalData>::makeUnique(
-          factory_context.serverFactoryContext().threadLocal())),
-      stats_(generateCryptoMbStats("cryptomb", factory_context.statsScope())) {
+    Server::Configuration::ServerFactoryContext& factory_context, IppCryptoSharedPtr ipp)
+    : api_(factory_context.api()),
+      tls_(ThreadLocal::TypedSlot<ThreadLocalData>::makeUnique(factory_context.threadLocal())),
+      stats_(generateCryptoMbStats("cryptomb", factory_context.scope())) {
```

Comment:
I think this isn't equivalent because generally the stats scope for the TransportSocketFactoryContext had a custom scope.  On the server side isn't it created in the listener via

scope_(server.stats().createScope("listener_manager.")), stats_(generateStats(*scope_)),?

This would change the name of all the generated stats and if so I think it's a breaking change we can't submit :-(
/wait

Reply:
Yes, I noted this change in the description. I could not locate any other private key method provider except the two contrib extensions `cryptomb` and `qat`, so the impact to stats is only limited to those two AFAICT. I don't think these extensions actually depend on the upstream/downstream context - so there's no value in keeping the association to the upstream/downstream scope in the stats. 

@giantcroc @soulxu - Is it OK to move the stats root for these extensions to server level?

Reply:
Unfortunately it's not just the contrib extensions here, where it's not just the owners but the end users we care about, it's also anyone who has their own extension points.

OOC are the factories created before runtime flags are assessed such that we could successfully restart-flag guard this?  You should be able to verify with an e2e test.  If we can runtime guard I'd suggest a runtime guard with prior structure and switching to server scope ahead of the cleanup.  As much as I prefer this clean up and hope we can move to it I just want to make sure we don't break anyone's monitoring metrics without recourse.

If we can't runtime guard I think we should have a quick discussion on maintainers as I don't think we have any policy around changing stats prefixes - I know we wouldn't for core code but in this case it's not clear if folks even monitor the stats so may be ok?  In that case we'd definitely want release notes.

/wait-any